packages/flutter/lib/src/rendering/wrap.dart

```diff
@@ -166,8 +166,58 @@ class _RunMetrics {
   }
 }
 
+
+/// How the child is inscribed into the available space.
+///
+/// See also:
+///
+///  * [Wrap], the wrap widget.
+///  * [Wrapped], the widget to set the [WrapFit].
+enum WrapFit {
+  /// Determines whether to place the child in the current or the
+  /// next run, depending on its minimum intrinsic size.
+  ///
+  /// The child is stretched to fill the entire run
+  /// (this does not apply if the [Wrap] is unconstrained).
+  ///
+  /// This is the default fit of the [Wrapped] widget.
+  ///
+  /// This setting can be expensive (see [RenderBox.getMinIntrinsicWidth]).
+  runTight(true),
+
+  /// Determines whether to place the child in the current or the
+  /// next run, depending on its minimum intrinsic size.
+  ///
+  /// The child is stretched to fill the entire run if the next child starts a
+  /// new run (this does not apply if the [Wrap] is unconstrained).
+  ///
+  /// This setting can be expensive (see [RenderBox.getMinIntrinsicWidth]).
+  runMaybeTight(false),
+
+  /// Determines whether to place the child in the current or the
+  /// next run, depending on its minimum intrinsic size.
+  ///
+  /// This setting can be expensive (see [RenderBox.getMinIntrinsicWidth]).
+  runLoose(false),
+
+  /// The child is placed in a new run and stretched to fill the
+  /// available space.
+  ///
+  /// (This does not apply if the [Wrap] is unconstrained.)
+  tight(true);
+
+  const WrapFit(this.isTight);
+
+  /// `true` if the [WrapFit] always forces the child to fill the assigned run.
+  final bool isTight;
+}
+
 /// Parent data for use with [RenderWrap].
-class WrapParentData extends ContainerBoxParentData<RenderBox> { }
+class WrapParentData extends ContainerBoxParentData<RenderBox> {
+  /// How a wrapped child is inscribed into the available space of the current
+  /// or an empty run.
+  WrapFit? fit;
```

Comment:
@nate-thegrate I did remove `WrapFit.loose` in fc8793561e.
Do you think this variable should be private?

Reply:
I'll leave it public, since `Wrapped` belongs to `widgets/` and it matches the `FlexParentData`.
https://github.com/flutter/flutter/blob/27234a90b9bd50338b1481235597671b66094293/packages/flutter/lib/src/rendering/flex.dart#L132